src/storage.rs

```diff
@@ -1,91 +1,104 @@
+    pub chat_telegram_id: ChatId,
+    date: Timespec,
 }
 
-impl From<Message> for Item {
-    fn from(message: Message) -> Item {
-        Item {
-            chat_id: message.chat.id(),
-            message_id: message.id,
-            date: message.date as u64,
+impl From<TelegramMessage> for Message {
+    fn from(message: TelegramMessage) -> Message {
+        Message {
+            id: 0,
+            telegram_id: message.id,
+            chat_telegram_id: message.chat.id(),
+            date: Timespec::new(message.date, 0),
         }
     }
 }
 
-impl Ord for Item {
-    fn cmp(&self, other: &Item) -> Ordering {
-        other.date.cmp(&self.date)
-    }
-}
-
-impl PartialOrd for Item {
-    fn partial_cmp(&self, other: &Item) -> Option<Ordering> {
-        Some(self.cmp(other))
-    }
-}
-
-impl PartialEq for Item {
-    fn eq(&self, other: &Item) -> bool {
-        self.date == other.date
-    }
-}
-
-fn now() -> u64 {
-    SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .unwrap()
-        .as_secs()
-}
-
 pub struct Storage {
-    heap: BinaryHeap<Item>,
-    lifetime: u64,
+    connection: Connection,
+    lifetime: u32,
 }
 
 impl Storage {
-    pub fn new(lifetime: u64) -> Storage {
+    pub fn new(path: &String, lifetime: u32) -> Storage {
+        debug!("Looking for storage at `{}`", path);
+        let connection = Connection::open_with_flags(
+            Path::new(path),
+            OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE,
+        ).unwrap();
+        connection
+            .execute(
+                "CREATE TABLE IF NOT EXISTS message (
+                    id                  INTEGER PRIMARY KEY,
+                    telegram_id         INTEGER NOT NULL,
+                    chat_telegram_id    INTEGER NOT NULL,
+                    date                TEXT NOT NULL
+                )",
+                &[],
+            ).expect("Troubles during table creation");
         Storage {
-            heap: BinaryHeap::new(),
+            connection,
             lifetime,
         }
     }
 
-    pub fn add(&mut self, message: Message) {
-        self.heap.push(message.into())
+    pub fn add(&self, message: TelegramMessage) {
+        let message: Message = message.into();
+        debug!("Inserting message {:?}", message);
+        let telegram_id: i64 = message.telegram_id.into();
+        let chat_telegram_id: i64 = message.chat_telegram_id.into();
+        self.connection
+            .execute(
+                "INSERT INTO message (telegram_id, chat_telegram_id, date)
+                    VALUES (?1, ?2, ?3)",
+                &[&telegram_id, &chat_telegram_id, &message.date],
+            ).expect("Troubles during message insertion");
     }
 
-    pub fn clean(&mut self) -> OutdateIter {
-        OutdateIter {
-            heap: &mut self.heap,
-            date: now() - self.lifetime,
-        }
+    pub fn clean(&mut self) -> Result<Vec<Message>> {
```

Comment:
Indeed, you can use iterator here (that commit transaction in `Drop::drop` impl), but for simplification `Vec<Message>` is enough.

Reply:
> that commit transaction in Drop::drop impl

This may be great excerise for me but I think it would be overcomplication.